ethportal-api/src/types/execution/transaction.rs

```diff
@@ -346,31 +440,37 @@ impl<'de> Deserialize<'de> for ToAddress {
         let s: Option<String> = Deserialize::deserialize(deserializer)?;
         match s {
             None => Ok(Self::Empty),
-            Some(val) => Ok(Self::Exists(H160::from_slice(
+            Some(val) => Ok(Self::Exists(Address::from_slice(
                 &hex_decode(&val).map_err(serde::de::Error::custom)?,
             ))),
         }
     }
 }
 
 impl Encodable for ToAddress {
-    fn rlp_append(&self, s: &mut RlpStream) {
+    fn encode(&self, out: &mut dyn bytes::BufMut) {
         match self {
             ToAddress::Empty => {
-                s.append_internal(&"");
+                out.put_u8(EMPTY_STRING_CODE);
             }
             ToAddress::Exists(addr) => {
-                s.append_internal(addr);
+                addr.0.encode(out);
             }
         }
     }
 }
 
 impl Decodable for ToAddress {
-    fn decode(rlp: &Rlp<'_>) -> Result<Self, DecoderError> {
-        match rlp.is_empty() {
-            true => Ok(ToAddress::Empty),
-            false => Ok(ToAddress::Exists(rlp::decode(rlp.as_raw())?)),
+    fn decode(buf: &mut &[u8]) -> alloy_rlp::Result<Self> {
+        if let Some(&first) = buf.first() {
+            if first == EMPTY_STRING_CODE {
+                buf.advance(1);
```

Comment:
Why are we advancing the `buf` here? I'm just not sure that there's a test for this code path, or did you find one and this was required to get it passing?

Reply:
I think it is less error prone as well